terraform/setup/setup.tf

```diff
@@ -0,0 +1,150 @@
+# in this module, we create the necessary common resources for the integ-tests, this setup module will only need to be executed once.
+# vpc, iam role, security group, the number of those resources could be limited, creating them concurrently for every pr would trigger throttling issue.
```

Comment:
Since you use default VPC you can clarify that here, I assumed from this comment you were creating a VPC here

Reply:
Though maybe you want to consider just creating a VPC here, so it's easier to manage e.g., private subnets

https://github.com/aws-samples/aws-xray-sdk-with-opentelemetry-sample/blob/both-otel-and-xray/scripts/terraform/network.tf#L75

Reply:
yes, i will probably create a vpc for integ-test when i do ecs and fargate if the default one is not easy to use.